columnphysics/icepack_shortwave.F90

```diff
@@ -4742,750 +4728,737 @@ subroutine compute_dEdd_5bd (klev,  klevp, &
+         do nsky = 1, 2 ! loop for both direct beam and diffuse beam
+         if (nsky == 1) then ! direct incident
+            do k = 0, nslyr
+               ! use top rsnw, rhosnw for snow ssl and rest of top layer
+               ksnow = k - min(k-1,0)
+               if (rsnw(ksnow) <= rsnw_snicar_min) then
+                  ks = ext_cff_mss_ice_drc(ns,1)
+                  ws = ss_alb_ice_drc     (ns,1)
+                  gs = asm_prm_ice_drc    (ns,1)
+               elseif (rsnw(ksnow) >= rsnw_snicar_max) then
+                  ks = ext_cff_mss_ice_drc(ns,nmbrad_snicar)
+                  ws = ss_alb_ice_drc     (ns,nmbrad_snicar)
+                  gs = asm_prm_ice_drc    (ns,nmbrad_snicar)
+               elseif (ceiling(rsnw(ksnow)) - rsnw(ksnow) < 1.0e-3_dbl_kind) then
+                  ! radius = 30 --> nr = 1 in SNICAR table      ! NOTE
+                  nr = ceiling(rsnw(ksnow)) - 30 + 1            ! hardwired min radius = 30
                   ks = ext_cff_mss_ice_drc(ns,nr)
-                  ws = ss_alb_ice_drc(ns,nr)
-                  gs = asm_prm_ice_drc(ns,nr)
-              else ! linear interpolation in rsnw
-                   ! radius = 30 --> nr = 1 in SNICAR table
-                  nr = ceiling(rsnw(ksnow)) - 30 + 1
-                  delr = (rsnw(ksnow) - floor(rsnw(ksnow))) / &
-                         (ceiling(rsnw(ksnow)) - floor(rsnw(ksnow)))
-                  ks = ext_cff_mss_ice_drc(ns,nr-1)*(delr) + &
-                       ext_cff_mss_ice_drc(ns,nr)*(c1-delr)
-                  ws = ss_alb_ice_drc(ns,nr-1)*(delr) + &
-                       ss_alb_ice_drc(ns,nr)*(c1-delr)
-                  gs = asm_prm_ice_drc(ns,nr-1)*(delr) + &
-                       asm_prm_ice_drc(ns,nr)*(c1-delr)
-              endif
-              tau(k) = (ks*rhosnw(ksnow) + kabs_chl_5bd(ns,k))*dzk(k)
-              w0(k) = (ks*rhosnw(ksnow))/(ks*rhosnw(ksnow) + kabs_chl_5bd(ns,k)) * ws
-              g(k)   = gs
-
-              !write(warning, *) "sky, k, tau, w0, g =", nsky, k, tau(k), w0(k), g(k)
-              !write(warning, *) "ns, ks, kabs_chl_5bd(ns,k), ", ns, ks, kabs_chl_5bd(ns,k)
-              !call add_warning(warning)
-              !print *, "rsnw(ksnow)", rsnw(ksnow)
-              !print *, "sky, k, tau, w0, g =", nsky, k, tau(k), w0(k), g(k)
-              !print *, "ns, ks, kabs_chl_5bd(ns,k), ",ns, ks, kabs_chl_5bd(ns,k)
-
+                  ws = ss_alb_ice_drc     (ns,nr)
+                  gs = asm_prm_ice_drc    (ns,nr)
+               else ! linear interpolation in rsnw
+                  nr = ceiling(rsnw(ksnow)) - 30 + 1            ! hardwired min radius = 30
+                  delr = (rsnw(ksnow) - floor(rsnw(ksnow))) & ! hardwired delta radius = 1 in table
+                       / (ceiling(rsnw(ksnow)) - floor(rsnw(ksnow))) ! denom always = 1?
+                  ks = ext_cff_mss_ice_drc(ns,nr-1)*    delr &  ! echmod: BUG delr factors opposite 3bd case
```

Comment:
@njeffery the bug is here, line 4889 of icepack_shortwave.F90 in hash 64ab4a16d0872d49ed40a96f7d36a8a281fc6cf3

Reply:
Thanks @eclare108213 .   I'll set up some runs in E3SM to test the impact of this bug.